### PR TITLE
python3Packages.python-lsp-black: init at 1.0.0

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change
I need this package, the other one `pyls-black` uses `python-language-server` which is being deprecated.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/python-lsp-black/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  checkInputs = [ pytestCheckHook ];
+
+  propagatedBuildInputs = [ black toml python-lsp-server ];
+
+  meta = with lib; {
+    homepage = "https://github.com/python-lsp/python-lsp-black";
+    description = "python-lsp-server plugin that adds support to black autoformatter";
+    license = licenses.mit;
+    maintainers = [ yourfavouriteuncle ];
```
Comment (FRidh) [inline pkgs/development/python-modules/python-lsp-black/default.nix:32]: you need to add yourself in the maintainers list

**Review by FRidh — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -12058,6 +12058,10 @@
       fingerprint = "FD0A C425 9EF5 4084 F99F 9B47 2ACC 9749 7C68 FAD4";
     }];
   };
+  yourfavouriteuncle = {
+    name = "yourfavouriteuncle";
+    email = "facata@cock.li";
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:12063]:
```suggestion
    email = "facata@cock.li";
    github = "yourfavouriteuncle";
    githubId = 75366929;
```

**Comment by yourfavouriteuncle** [comment]
Thanks for reviewing my changes quickly and apologies for the minor mistakes.

File: pkgs/development/python-modules/python-lsp-black/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  src = fetchFromGitHub {
+    owner = "python-lsp";
+    repo = "python-lsp-black";
+    rev = "v${version}";
+    sha256 = "1blxhj70jxb9xfbd4dxqikd262n6dn9dw5qhyml5yvdwxbv0bybc";
+  };
+
+  disabled = !isPy3k;
+
+  checkInputs = [ pytestCheckHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-lsp-black/default.nix:24]:
```suggestion
  checkInputs = [ pytestCheckHook ];

  pythonImportsCheck = [ "pylsp_black" ];  
```

File: pkgs/development/python-modules/python-lsp-black/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "python-lsp";
+    repo = "python-lsp-black";
+    rev = "v${version}";
+    sha256 = "1blxhj70jxb9xfbd4dxqikd262n6dn9dw5qhyml5yvdwxbv0bybc";
+  };
+
+  disabled = !isPy3k;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/python-lsp-black/default.nix:22]:
```suggestion
  disabled = pythonOlder "3.6";
```

https://github.com/python-lsp/python-lsp-black/blob/master/setup.cfg#L18

File: pkgs/development/python-modules/python-lsp-black/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, black
+, toml
+, pytestCheckHook
+, python-lsp-server
+, isPy3k
```
Comment (georgyo) [inline pkgs/development/python-modules/python-lsp-black/default.nix:8]: `isPy3k` is unused

File: pkgs/development/python-modules/python-lsp-black/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "python-lsp";
+    repo = "python-lsp-black";
+    rev = "v${version}";
+    sha256 = "1blxhj70jxb9xfbd4dxqikd262n6dn9dw5qhyml5yvdwxbv0bybc";
+  };
+
+  disabled = pythonOlder "3.6";
```
Comment (georgyo) [inline pkgs/development/python-modules/python-lsp-black/default.nix:22]: `pythonOlder` is missing from the function definition.

**Review by georgyo — CHANGES_REQUESTED** [review]

**Comment by georgyo** [comment]
Actually, python-lsp-black has already been merged in #131060

**Pull request closed without merging**